libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsStorage.java

```diff
@@ -3,5 +3,187 @@
+import com.microsoft.bot.restclient.serializer.JacksonAdapter;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class BlobsStorage implements Storage {
+    private final JacksonAdapter jacksonAdapter;
+    private final BlobContainerClient containerClient;
+
+    public BlobsStorage(String dataConnectionString, String containerName) {
+        if (StringUtils.isBlank(dataConnectionString)) {
+            throw new IllegalArgumentException("dataConnectionString is required.");
+        }
+
+        if (StringUtils.isBlank(containerName)) {
+            throw new IllegalArgumentException("containerName is required.");
+        }
+
+        jacksonAdapter = new JacksonAdapter();
+        containerClient = new BlobContainerClientBuilder()
+                            .connectionString(dataConnectionString)
+                            .containerName(containerName)
+                            .buildClient();
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(String[] keys) {
+        if (keys == null) {
+            throw new IllegalArgumentException("The 'keys' parameter is required.");
+        }
+
+        for (String key : keys) {
+            String blobName = GetBlobName(key);
+            BlobClient blobClient = containerClient.getBlobClient(blobName);
+            blobClient.delete();
+        }
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @Override
+    public CompletableFuture<Map<String, Object>> read(String[] keys) {
+        if (keys == null) {
+            throw new IllegalArgumentException("The 'keys' parameter is required.");
+        }
+
+        initialize();
+
+        Map<String, Object> items = new HashMap<>();
+
+        for (String key : keys) {
+            try {
+                String blobName = GetBlobName(key);
+                BlobClient blobClient = containerClient.getBlobClient(blobName);
+                items.put(key, innerReadBlob(blobClient).get());
+            } catch (HttpResponseException ex) {
+                if (ex.getResponse().getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                    continue;
+                }
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+            }
+
+        }
+        return CompletableFuture.supplyAsync(() -> items);
+    }
+
+    public CompletableFuture<Void> write(Map<String, Object> changes) {
+        if (changes == null) {
+            throw new IllegalArgumentException("The 'changes' parameter is required.");
+        }
+
```

Comment:
Are you missing the create if not exists, check [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsStorage.cs#L154)?

Reply:
Yes, we need to find a good way check this, related to Lee's feedback